src/Attestation/Attestation/Models/AttestationManagementClient.cs

```diff
@@ -74,16 +81,55 @@ public PSAttestation GetAttestation(string attestationName, string resourceGroup
         }
         public void DeleteAttestation(string attestationName, string resourceGroupName)
         {
-            if (string.IsNullOrWhiteSpace(attestationName))
+            if (string.IsNullOrEmpty(attestationName))
             {
                 throw new ArgumentNullException("attestationName");
             }
 
-            if (string.IsNullOrWhiteSpace(resourceGroupName))
+            if (string.IsNullOrEmpty(resourceGroupName))
             {
                 throw new ArgumentNullException("resourceGroupName");
             }
             attestationClient.AttestationProviders.Delete(resourceGroupName, attestationName);
         }
+
+        public X509Certificate2Collection GetX509CertificateFromPEM(string pemString, string section)
+        {
+            X509Certificate2Collection certificateCollection = new X509Certificate2Collection();
+            var header = String.Format("-----BEGIN {0}-----", section);
+            var footer = String.Format("-----END {0}-----", section);
+
+            var start = 0;
+            var end = 0;
+            while (true)
+            {
+                start = pemString.IndexOf(header, StringComparison.Ordinal);
+
+                if (start < 0)
+                    break;
+                start += header.Length;
+                end = pemString.IndexOf(footer, start, StringComparison.Ordinal) - start;
```

Comment:
`end` means the length of this section. So it will be better that rename it to `lengthOfSection`.

Reply:
replace end with lengthOfSection